### PR TITLE
Vault-21960: Add docker tests for reloading seal configuration on SIGHUP

### PR DESCRIPTION
**Pull request by rculpepper**

### CODE REVIEW

File: vault/external_tests/seal_binary/seal_sighup_test.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+			expectedSealTypes: []string{
+				"transit",
+				"transit",
+			},
+		},
+		"more than one seal fails": {
+			sealStanzas: []string{
+				fmt.Sprintf(transitStanza, firstTransitKeyConfig, 1, "false"),
+				fmt.Sprintf(transitStanza, firstTransitKeyConfig, 1, "false") + "," +
+					fmt.Sprintf(transitStanza, secondTransitKeyConfig, 2, "false"),
```
Comment (victorr) [inline vault/external_tests/seal_binary/seal_sighup_test.go:95]: I am curious about this test case. Why doesn't going from autoseal to multiseal work?

**Review by victorr — APPROVED** [review]
This code is very good, well done!

**Comment by victorr** [comment]
BTW, I will have some suggestions for this PR on https://github.com/hashicorp/vault-enterprise/pull/5048.

File: vault/external_tests/seal_binary/seal_sighup_test.go (hunk 1 above)
Comment (rculpepper) [reply vault/external_tests/seal_binary/seal_sighup_test.go:95]: These tests only run for non-enterprise builds, so configuring multiple enabled seals will fail

**Pull request merged**